api/nvidia/v1/clusterpolicy_types.go

```diff
@@ -2094,3 +2099,36 @@ func (c *CCManagerSpec) IsEnabled() bool {
 	}
 	return *c.Enabled
 }
+
+type ConfigWithName interface {
+	GetName() string
+}
+
+// GetConfigMapName returns the config's name if it's non-empty and differs from defaultName,
+// otherwise returns defaultName. The boolean indicates whether a custom name was used.
+func GetConfigMapName[T ConfigWithName](config T, defaultName string) (string, bool) {
+	if config.GetName() != "" && config.GetName() != defaultName {
+		return config.GetName(), true
+	}
+	return defaultName, false
+}
+
+// safeGetName safely gets the name from a pointer, handling nil cases.
+func safeGetName[T any](ptr *T, getName func(T) string) string {
```

Comment:
Can we import this method from `k8s.io/utils` instead? 

https://github.com/kubernetes/utils/tree/master/pointer

Reply:
Yes, this looks good to me. Thanks!